src/System.Net.Sockets/tests/FunctionalTests/NetworkStreamTest.cs

```diff
@@ -85,11 +639,14 @@ public class NetworkStreamTest
                 // Copying to non-writable stream
                 Assert.Throws<NotSupportedException>(() => { stream.CopyToAsync(new MemoryStream(new byte[0], writable: false)); });
 
-                // Copying after disposing the stream
+                // Already canceled
+                Assert.Equal(TaskStatus.Canceled, stream.CopyToAsync(new MemoryStream(new byte[1]), 1, new CancellationToken(canceled: true)).Status);
+
+                // Copying while and then after disposing the stream
+                Task copyTask = stream.CopyToAsync(new MemoryStream());
                 stream.Dispose();
+                await Assert.ThrowsAsync<IOException>(() => copyTask);
```

Comment:
@geoffkizer, this is failing on some of the unix runs, with no exception being thrown, which means the copy ends but without any exception.  That seems likely to be a bug in the implementation.  I'll open an issue and disable this for Unix for now, unless you think otherwise.